src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -253,21 +253,24 @@ protected override async Task<HttpResponseMessage> SendAsync (HttpRequestMessage
 
 			cancellationToken.Register (() => {
 				RemoveInflightData (dataTask);
+				dataTask?.Cancel ();
```

Comment:
not sure why this would be needed ?!?

The call to `RemoveInflightData` already calls `Cancel` on `dataTask` on line [172](https://github.com/xamarin/xamarin-macios/pull/5334/files#diff-7f619be41b43eaaed811ff8b330f769cL172)

Reply:
Because it would not already be in the dictionary of in-flight data to cancel if I read it right

Reply:
No, it has nothing to do with the dictionary (it's not inside the block).
This line is always executed so this is a _no op_

If you get different results it's because the test case is not very consistent in recreating the issue.

Reply:
Sorry You are right, I miss read the code. This line can be removed.